File: usermanagement/src/main/java/md/utm/fcim/usermanagement/webservice/UserController.java
package md.utm.fcim.usermanagement.webservice;

import org.springframework.stereotype.Component;

import javax.ws.rs.GET;
import javax.ws.rs.Path;

@Component
@Path("/api/user")
public interface UserController {

    @GET
    @Path("/")
    String test();
}

File: usermanagement/src/main/kotlin/md/utm/fcim/usermanagement/webservice/UserController.kt
package md.utm.fcim.usermanagement.webservice

import md.utm.fcim.usermanagement.webservice.view.UserView
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/api/user/")
interface UserController {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun retrieve(): Response

    @GET
    @Path("{id}")
    @Produces(MediaType.APPLICATION_JSON)
    fun retrieveById(@PathParam("id") id: Long): Response

    @POST
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    fun create(userView: UserView): Response
}
